iothub/service/src/Messaging/MessagingClient.cs

```diff
@@ -0,0 +1,379 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Common;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Transport types supported by MessagingClient.
+    /// </summary>
+    /// <remarks>
+    /// Amqp and Amqp over WebSocket only.
+    /// </remarks>
+    public enum TransportType
+    {
+        /// <summary>
+        /// Advanced Message Queuing Protocol transport.
+        /// </summary>
+        Amqp,
+
+        /// <summary>
+        /// Advanced Message Queuing Protocol transport over WebSocket only.
+        /// </summary>
+        Amqp_WebSocket_Only
```

Comment:
```suggestion
        Amqp_WebSocket
```

The "Only" suffix is a relic of when we wanted to implement fallback protocols (for example, try AMQP, but if that doesn't work, try AMQP_WS). We don't support fallback protocols in v2, so we can simplify this name